### PR TITLE
Overhaul port configuration in deployment

### PR DESCRIPTION
**Pull request by patrickkwang**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,6 +9,4 @@ ADD ./app ./app
 ADD ./main.sh ./main.sh
 
 # set up entrypoint
-CMD ["./main.sh"]
-ARG PORT=8080
-EXPOSE $PORT
\ No newline at end of file
+ENTRYPOINT ["./main.sh"]
```
Comment (PhillipsOwen) [inline Dockerfile:12]: this is what happens in windows if you dont explicitly use bash on the ENTRYPOINT in the first element when starting the container.

/usr/bin/env: â€˜bash\râ€™: No such file or directory

File: main.sh [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+    -p|--port)
+    PORT="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    *)    # unknown option
+    POSITIONAL+=("$1") # save it in an array for later
+    shift # past argument
+    ;;
+esac
```
Comment (PhillipsOwen) [inline main.sh:21]: i dont think i see the benefit of all this. i personally believe that the deploy configuration should be the lowest common denominator.

i will have to try this in k8s. right now the deployment provides the port via an env variable.

File: Dockerfile (hunk 1 above)
Comment (PhillipsOwen) [reply Dockerfile:12]: and the same thing happens in k8s.

Comment (patrickkwang) [reply Dockerfile:12]: I've managed to deploy this with minikube on MacOS without issues. I also can't find any indication on the internet of others having this issue with bash scripts and k8s, so I have to suspect that something else is going on in your particular case. Can you provide more details about what you're trying: host OS, docker/k8s versions, etc.?

File: main.sh (hunk 2 above)
Comment (patrickkwang) [reply main.sh:21]: You should be able to provide `"command": ["--port", "XXXX"]` in the k8s yaml file if you need the container to use a particular port: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/, or my feeling is you should be able to forward the port wherever you need it at a higher level. But I'm just starting to play around with k8s.

File: Dockerfile (hunk 1 above)
Comment (PhillipsOwen) [reply Dockerfile:12]: windows 10 and docker desktop (both latest versions) integrated with PyCharm. 

i will see what happens when i create the image on WSL.

Comment (PhillipsOwen) [reply Dockerfile:12]: it appears to be a dos vs. linux line endings issue. WSL reports: "/usr/bin/env: ‘bash\r’: No such file or directory"

Comment (PhillipsOwen) [reply Dockerfile:12]: converting main.sh to use unix line endings fixed the issue

**Pull request merged**